File: zs-leetcode-go/MajorityElement/MajorityElement.go
package MajorityElement


func majorityElement(nums []int) int {
    times := 1
    aim := nums[0]
    for _, item := range nums[1:] {
        if item == aim {
            times++
        } else {
            times--
        }
        if times == 0 {
            aim = item
            times = 1
        }
    }
    return aim
}

File: sjq-leetcode-C++/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int> mapNums;
        int res=0;
        for(int i=0; i< nums.size(); i++){
            mapNums[nums[i]]++;
            if(mapNums[nums[i]]>nums.size()/2)
                res=nums[i];
        }
        return res;
    }
};
